app/components/challenge-workspace-buttons.js

```diff
@@ -90,6 +91,12 @@ export default Component.extend({
 
     guardarSolucion() {
       let activityName = this.get("actividad.nombre");
+      if( activityName === undefined ) {
+        const challengeJson = this.storage.getImportedChallenge()
+        if (challengeJson)
+          activityName = challengeJson.titulo || 'sin titulo'
+      }
```

Comment:
Genial! El problema con esto es que si todo da undefined, el nombre de archivo sigue siendo undefined, porque el segundo if no tiene else. Una propuesta:

```suggestion
      let activityName = this.get("actividad.nombre") || this.storage.getImportedChallenge()?.titulo || "Sin título"
```

Reply:
aaaaaaaaaahhhhh... perdon. Yo entendí que, como el guardarSolucion esta en Ember haciendo lo que hace, era por ahi que venia el cambiazo... Y no sabia si haciendo lo que propones en React se resolvia solito...
Respecto a tu sugerencia, no la planteé asi porque entendia que solo daba undefined cuando se usaba esta nueva logica del desafio viajando desde Ract, no pensé que podria existir fuera de esa situacion, de lo que entendí leyendo el issue.
Y ademas, dudaba si podiamos usar la funcion ()? a partir de this.storage y que al hacerlo me diera un error peor en ember por alguna __buena__ otra razon. Es por eso que lo anidé cambiandolo solo cuando me aseguraba que podia obtener el dato que aunque en blanco, (en el ||) pero que existiera challengeJson.
Commiteo tu sugerencia